syntax = "proto3";

package perception;

// Family of the april tag being used
// Specifies properties of the AprilTag like number of codes, minimum hamming distance between codes etc.
enum AprilTagFamily {
    AprilTag36h11 = 0;
    AprilTag36h10 = 1;
    AprilTag25h9 = 2;
    AprilTag25h7 = 3;
    AprilTag16h5 = 4;
}

message AprilTagConfig {
    // Family of the april tag being used
    AprilTagFamily aprilTagFamily = 1;

    // Side lenth of the outer black square
    float sideLengthInMeters = 2;

    // Width of the black border in bits
    uint32 border = 3;
}
